packages/block-editor/src/components/use-editor-feature/index.js

```diff
@@ -1,23 +1,51 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
 /**
  * WordPress dependencies
  */
 import { useSelect } from '@wordpress/data';
 
+/**
+ * Internal dependencies
+ */
+import { useBlockEditContext } from '../block-edit';
+
 /**
  * Hook that retrieves the setting for the given editor feature.
+ * It works with nested objects using by finding the value at path.
  *
- * @param {string} featureName The name of the feature.
+ * @param {string} featurePath  The path to the feature.
+ * @param {*}      defaultValue Default value to return if not
+ *                              explicitly defined.
  *
  * @return {any} Returns the value defined for the setting.
+ *
+ * @example
+ * ```js
+ * const isEnabled = useEditorFeature( 'typography.dropCap', false );
+ * ```
  */
-export default function useEditorFeature( featureName ) {
+export default function useEditorFeature( featurePath, defaultValue ) {
```

Comment:
I'm thinking that this component merges data from a few sources: block settings, editor settings, block.json, theme.json, filters, etc. Do we need to take a default value as well?

Reply:
There are filters, 3rd party plugins, 3rd party blocks, you can't guarantee that the value is provided. In addition, it aligns with `hasBlockSupport` API methods from `@wordpress/blocks`. 

On the other hand, it might be more convenient to avoid it but then we need to ensure that the block editor loads all the defaults when there is no server.

Reply:
We agreed to remove the `defaultValue` param for now.